go/updater/trainingJobUpdater.go

```diff
@@ -48,34 +55,417 @@ type TrainingJobUpdater struct {
+		log.Error("delete trainer replicaset error: ", err.Error())
+		fault = true
+	}
+
+	log.Infof("Release trainer, namespace=%v name=%v", updater.job.Namespace, updater.job.Spec.Trainer.ReplicaSpec.Name)
+	if err := updater.releaseTrainer(); err != nil {
+		log.Error("release trainer  error: ", err.Error())
+		fault = true
+	}
+
+	log.Infof("End to delete TrainingJob namespace=%v name=%v", updater.job.Namespace, updater.job.Name)
+
+	if fault {
+		return fmt.Errorf("delete resource error namespace=%v name=%v", updater.job.Namespace, updater.job.Name)
+	}
+	return nil
+}
+
+func (updater *TrainingJobUpdater) createResource(tp padv1.TrainingResourceType) error {
+	resource := new(v1beta1.ReplicaSet)
+	switch tp {
+	case padv1.MASTER:
+		resource = updater.job.Spec.Master.ReplicaSpec
+	case padv1.PSERVER:
+		resource = updater.job.Spec.Pserver.ReplicaSpec
+	default:
+		return fmt.Errorf("unknow resource")
+	}
+	newResource, err := updater.kubeClient.ExtensionsV1beta1().ReplicaSets(updater.job.Namespace).Get(resource.Name, v1.GetOptions{})
+	if errors.IsNotFound(err) {
+		log.Infof("not found to create namespace=%v name=%v resourceName=%v", updater.job.Namespace, updater.job.Name, resource.Name)
+		newResource, err = updater.kubeClient.ExtensionsV1beta1().ReplicaSets(updater.job.Namespace).Create(resource)
+		if err != nil {
+			updater.status.Phase = padv1.TrainingJobPhaseFailed
+			updater.status.Reason = "Internal error; create resource error:" + err.Error()
+			return err
+		}
+	}
+	if newResource != nil {
+		ticker := time.NewTicker(confirmResourceTicker)
+		defer ticker.Stop()
+		for v := range ticker.C {
+			rs, err := updater.kubeClient.ExtensionsV1beta1().ReplicaSets(updater.job.Namespace).Get(resource.Name, v1.GetOptions{})
+			log.Infof("Current time %v runing pod is %v, resourceName=%v", v.String(), rs.Status.ReadyReplicas, resource.Name)
+			if err != nil && !errors.IsServerTimeout(err) && !errors.IsTooManyRequests(err) {
+				updater.status.Reason = "Internal error; create resource error:" + err.Error()
+				return err
+			}
+			if errors.IsServerTimeout(err) || errors.IsTooManyRequests(err) {
+				log.Warningf("Connect to kubernetes failed for reasons=%v, retry next ticker", err.Error())
+				continue
+			}
+			if *resource.Spec.Replicas == 0 {
+				return fmt.Errorf(" trainingjob is deleting, namespace=%v name=%v ", updater.job.Namespace, updater.job.Name)
+
+			}
+			if rs.Status.ReadyReplicas == *resource.Spec.Replicas {
+				log.Infof("Create resource done , namespace=%v name=%v resourceName=%v", updater.job.Namespace, updater.job.Name, resource.Name)
+				return nil
+			}
+		}
+	}
+	return nil
+}
+
+func (updater *TrainingJobUpdater) createTrainer() error {
+	newTrainer, err := updater.kubeClient.BatchV1().Jobs(updater.job.Namespace).Get(updater.job.Spec.Trainer.ReplicaSpec.Name, v1.GetOptions{})
+	if errors.IsNotFound(err) {
```

Comment:
The same as: https://github.com/PaddlePaddle/cloud/pull/634/files#r174057818

Reply:
Done. Thank you very much for your comments, and I have fixed all of the comments you mentioned above.